Code/Tools/FBuild/FBuildCore/Graph/ObjectListNode.cpp

```diff
@@ -349,14 +351,14 @@ bool ObjectListNode::CreateDynamicObjectNode( Node * inputFile, const DirectoryL
     // if source comes from a directory listing, use path relative to dirlist base
     // to replicate the folder hierearchy in the output
     AStackString<> subPath;
-    if ( dirNode )
+    if ( baseDir.GetLength() > 0 )
     {
-        const AString & baseDir = dirNode->GetPath();
-        ASSERT( PathUtils::PathBeginsWith( fileName, baseDir ) );
-        if ( PathUtils::PathBeginsWith( fileName, baseDir ) )
+		AString fullBaseDir;
+		NodeGraph::CleanPath(baseDir, fullBaseDir);
```

Comment:
As noted, by doing the CleanPath during parsing, we avoid doing it at build time.  I'll take care of this as part of the integration.